packages/daemon/src/service/pipeline.ts

```diff
@@ -48,6 +48,220 @@ function execAsync(cmd: string, opts: ExecOptions): Promise<string> {
   });
 }
 
+interface RunnerOptions {
+  job: JobEntity;
+  pipeline: PipelineEntity;
+  plugins: Partial<Record<PluginTypeI, PluginInfo>>;
+  tracer: Tracer;
+  runnable: PluginRunnable;
+  datasetRoot: string;
+}
+
+interface ModelResult {
+  model: RunnableResponse;
+  plugin: PluginPackage;
+}
+
+interface RunnerResult {
+  output: any;
+  dataset: any;
+  modelPath: string;
+  modelPlugin: PluginPackage;
+  dataProcess: PluginPackage;
+  datasetProcess: PluginPackage;
+}
+
+/**
+ * job runner
+ * handle the pipeline running: check pipeline config, load and run plugins, dispatch pipeline running event.
+ */
+export class JobRunner {
```

Comment:
Shall we move this to https://github.com/alibaba/pipcook/blob/8c43574df00d2fa59f7d0b15c00ca417af3641a6/packages/daemon/src/runner/helper.ts?

Reply:
how about move runner into `utils/job-runner.ts`?